src/battle_regulation.c

```diff
@@ -0,0 +1,276 @@
+#include "battle_regulation.h"
+
+#include "global.h"
+
+#include "constants/items.h"
+
+#include "dex_mon_measures.h"
+#include "party.h"
+#include "pokemon.h"
+#include "unk_020290B4.h"
+
+static BOOL sub_02074C50(u16 *a0, u16 *a1, u16 *a2, int a3, int a4, int a5, int a6);
+
+PokedexData *sub_02074944(HeapID heapId) {
+    PokedexData *pokedex = PokedexData_Create(heapId);
+    PokedexData_LoadAll(pokedex, 0, heapId);
+    return pokedex;
+}
+
+void sub_0207495C(PokedexData *pokedex) {
+    PokedexData_UnloadAll(pokedex);
+    PokedexData_Delete(pokedex);
+}
+
+BOOL sub_0207496C(LinkBattleRuleset *ruleset, Pokemon *pokemon, PokedexData *pokedex) {
+    u16 species = GetMonData(pokemon, MON_DATA_SPECIES, NULL);
+    if (ruleset == NULL) {
+        return TRUE;
+    }
+
+    int rule = LinkBattleRuleset_GetRuleValue(ruleset, 2);
+    if (GetMonData(pokemon, MON_DATA_LEVEL, NULL) > rule) {
+        return FALSE;
+    }
+    if (GetMonData(pokemon, MON_DATA_IS_EGG, NULL)) {
+        return FALSE;
+    }
+
+    rule = LinkBattleRuleset_GetRuleValue(ruleset, 4);
+    if (rule == 0) {
+        if (species != GetEggSpecies(species)) {
+            return FALSE;
+        }
+    }
+
+    rule = LinkBattleRuleset_GetRuleValue(ruleset, 5);
+    if (rule != 0) {
+        int height = PokedexData_GetHeight(pokedex, species);
+        if (rule > 0) {
+            if (height < rule) {
+                return FALSE;
+            }
+        } else {
+            if (height > -rule) {
+                return FALSE;
+            }
+        }
+    }
+
+    rule = 10 * LinkBattleRuleset_GetRuleValue(ruleset, 7);
+    if (rule != 0) {
+        int weight = PokedexData_GetWeight(pokedex, species);
+        if (rule > 0) {
+            if (weight < rule) {
+                return FALSE;
+            }
+        } else {
+            if (weight > -rule) {
+                return FALSE;
+            }
+        }
+    }
+
+    rule = LinkBattleRuleset_GetRuleValue(ruleset, 13);
+    if (rule == 0x8000) {
+        return SpeciesIsMythical(species) != TRUE;
+    }
+
+    rule = LinkBattleRuleset_GetRuleValue(ruleset, 11);
+    if (rule == 0) {
+        if (MonCheckFrontierIneligibility(pokemon)) {
+            return FALSE;
+        }
+    }
+
+    return TRUE;
+}
+
+int sub_02074A6C(LinkBattleRuleset *a0, Party *party, PokedexData *pokedex, u8 *selectedOrder) {
```

Comment:
this looks like it's checking for specific rule violations and returning a specified error const?

Reply:
also a0 -> ruleset

Reply:
Handled in #332